app/models/spree/role_decorator.rb

```diff
@@ -1,8 +1,13 @@
 Spree::Role.class_eval do
 
-  has_and_belongs_to_many :permissions, join_table: 'spree_roles_permissions', class_name: 'Spree::Permission'
+  has_many :roles_permission_sets, dependent: :destroy
+  has_many :permission_sets, through: :roles_permission_sets
+  has_many :permissions, through: :permission_sets
+
+  has_and_belongs_to_many :legacy_permissions, join_table: 'spree_roles_permissions', class_name: 'Spree::Permission'
```

Comment:
@nimish-mehta Lets add a comment for deprecating this legacy permission association.